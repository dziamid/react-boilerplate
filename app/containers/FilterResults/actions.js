/*
 *
 * FilterResults actions
 *
 */

import {
  DEFAULT_ACTION,
  SET_SELECTED_TITLE,
} from './constants';

export function defaultAction() {
  return {
    type: DEFAULT_ACTION,
  };
}

export function setSelectedTitle(titleId) {
  return {
    type: SET_SELECTED_TITLE,
    titleId
  };
}
