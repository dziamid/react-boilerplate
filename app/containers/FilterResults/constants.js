/*
 *
 * FilterResults constants
 *
 */

export const DEFAULT_ACTION = 'app/FilterResults/UPDATE_TITLE';
export const SET_SELECTED_TITLE = 'app/FilterResults/SET_SELECTED_TITLE';
