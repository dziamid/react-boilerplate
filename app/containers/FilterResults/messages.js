/*
 * FilterResults Messages
 *
 * This contains all the text for the FilterResults component.
 */
import { defineMessages } from 'react-intl';

export default defineMessages({
  select: {
    id: 'app.containers.FilterResults.select',
    defaultMessage: 'Select',
  },
  jobTitle: {
    id: 'app.containers.FilterResults.jobTitle',
    defaultMessage: 'Job Title',
  },
  seniority: {
    id: 'app.containers.FilterResults.seniority',
    defaultMessage: 'Seniority',
  },
  addRelation: {
    id: 'app.containers.FilterResults.addRelation',
    defaultMessage: 'Add Relation',
  },
  total: {
    id: 'app.containers.FilterResults.totals',
    defaultMessage: 'Displaying [[filtered]] out of [[total]] Titles',
  },
});
