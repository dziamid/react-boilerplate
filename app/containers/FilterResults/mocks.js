import { getRandomInt } from '../../utils/mathUtils';

export function getResults() {
  const num = getRandomInt(1, 10);

  const data =
    [
    { jobTitle: 'web dev', seniority: 2, relations: 4 },
    { jobTitle: 'html dev', seniority: 1, relations: 2 },
    { jobTitle: 'dev manager', seniority: 5, relations: 6 },
    { jobTitle: 'IT guy', seniority: 2, relations: 1 },
    { jobTitle: 'web dev', seniority: 2, relations: 4 },
    { jobTitle: 'html dev', seniority: 1, relations: 2 },
    { jobTitle: 'dev manager', seniority: 5, relations: 6 },
    { jobTitle: 'IT guy', seniority: 2, relations: 1 },
    ];

  return data.splice(0, num);
}

