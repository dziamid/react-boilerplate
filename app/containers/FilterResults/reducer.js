/*
 *
 * FilterResults reducer
 *
 */

import { fromJS } from 'immutable';
import {
  DEFAULT_ACTION,
  SET_SELECTED_TITLE,
} from './constants';

const initialState = fromJS({
  results: [],
});

function FilterResultsReducer(state = initialState, action) {
  switch (action.type) {
    case DEFAULT_ACTION:
      return state;
    case SET_SELECTED_TITLE:
      return state.set('selectedTitle', action.titleId);
    default:
      return state;
  }
}

export default FilterResultsReducer;
