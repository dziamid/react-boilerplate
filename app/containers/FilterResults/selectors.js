import { createSelector } from 'reselect';

const titles = () => (state) => state.getIn(['titlesEditorRoot', 'titles']);
const query = () => (state) => state.getIn(['form', 'FilterParams', 'values', 'filter']); // todo: why separate state root? use titlesEditorRoot.filterParams
const selectedTitle = () => (state) => state.getIn(['titlesEditorRoot', 'selectedTitle']);
const relations = () => (state) => state.getIn(['titlesEditorRoot', 'relations']);

const results = () => createSelector(
  titles(),
  query(),
  (titles, query) => {
    return titles.filter(t => t.name.toLowerCase().includes((query || '').trim()));
  });


export {
  titles,
  results,
  selectedTitle,
  relations,
};
