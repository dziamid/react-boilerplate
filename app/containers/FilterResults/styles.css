.titleColumn {
  min-width: 250px;
  cursor: pointer;
  overflow: hidden;
  width: 50%;
}

.seniorityColumn {
  min-width: 150px;
  overflow: hidden;
}

.relationsColumn {
  width: 80px;
  text-align: center !important;
}
