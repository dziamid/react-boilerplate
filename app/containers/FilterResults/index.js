/*
 *
 * FilterResults
 *
 */

import React, { PropTypes, Component } from 'react';
import { selectFilterResults, selectResultsList } from './selectors';
import selectFilterParams from 'containers/FilterParams/selectors';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import styles from './styles.css';
import { createStructuredSelector } from 'reselect';

import { setSelectedTitle } from './actions';

import { connect } from 'react-redux';
import { Field, reduxForm, change, SubmissionError } from 'redux-form/immutable';
import Button from 'components/Button'

export class FilterResults extends Component { // eslint-disable-line react/prefer-stateless-function
  render() {
    const {
      results,
      updateResults,
      filterParams: {
        titles,
      },
      filterText,
      setSelectedTitle,
      selectedTitle
    } = this.props;

    const filteredTitles = titles ?
      titles.filter(t => t.title.toLowerCase().includes((filterText || '').trim()))
      : [];

    const
      total = (titles || []).length,
      filtered = filteredTitles.length;

    const statusLine = `Displaying ${filtered} out of ${total} Titles`;

    const getRowStyle = (id) => {
      if (selectedTitle === id) {
        return styles.selectedRow;
      } else {
        return styles.tableRow;
      }
    };

    const singleResult = t => (
      <div className={getRowStyle(t._id)}>
          <span className={styles.selectCol}>
            <Button
              raised
              onClick={() => setSelectedTitle(t._id)}>Select</Button>
          </span>
        <span className={styles.jobTitleCol}>
                {t.title}
          </span>
        <span className={styles.seniorityCol}>
                {t.seniority}
          </span>
        <span className={styles.relationCol}>
                {t.relations}
          </span>
      </div>
    );

    return (
      <div>
        <div>
          {statusLine}
        </div>
        <div className={styles.tableWrapper}>
          <div className={styles.tableHeader}>
          <span className={styles.selectHeader}>
            <FormattedMessage {...messages.select} />
          </span>
            <span className={styles.jobTitleHeader}>
            <FormattedMessage {...messages.jobTitle} />
          </span>
            <span className={styles.seniorityCol}>
            <FormattedMessage {...messages.seniority} />
          </span>
            <span className={styles.relationCol}>
            <FormattedMessage {...messages.addRelation} />
          </span>
          </div>

          {filteredTitles.map(t => singleResult(t))}
        </div>
      </div>
    );
  }
}

FilterResults.propTypes = {
  handleSubmit: PropTypes.func,
  reset: PropTypes.func,
  submitting: PropTypes.bool,
  dispatch: PropTypes.func,
  form: PropTypes.string,
};

const mapStateToProps = createStructuredSelector({
  // results: selectResultsList(),
  filterParams: selectFilterParams(),
  filterText: (state) => state.getIn(['form', 'FilterParams', 'values', 'filter']),
  selectedTitle: (state) => state.getIn(['titlesEditorRoot', 'filterResults', 'selectedTitle'])
});

function mapDispatchToProps(dispatch) {
  return {
    setSelectedTitle: (titleId) => dispatch(setSelectedTitle(titleId)),
  };
}

const validate = (values) => {
  const errors = {};

  if (!values.get('field1')) {
    errors.field1 = 'Required';
  }

  return errors;
};

const form = reduxForm({form: 'filterResults', validate})(FilterResults);
export default connect(mapStateToProps, mapDispatchToProps)(form);
