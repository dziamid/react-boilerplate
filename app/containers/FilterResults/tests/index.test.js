import React from 'react';
import FilterResults from '../index';
import { shallowToJson } from 'enzyme-to-json';
import { shallow } from 'enzyme';


describe('<FilterResults />', () => {
  it('should render show all companies correctly', () => {
    const wrapper = shallow(
      <FilterResults results={[]} />
    );

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
