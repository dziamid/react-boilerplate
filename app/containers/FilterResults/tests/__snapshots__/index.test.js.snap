exports[`<FilterResults /> should render show all companies correctly 1`] = `
<Table
  allRowsSelected={false}
  fixedFooter={true}
  fixedHeader={true}
  height="inherit"
  multiSelectable={false}
  onRowSelection={[Function]}
  selectable={true}>
  <TableHeader
    adjustForCheckbox={true}
    displaySelectAll={false}
    enableSelectAll={true}
    selectAllSelected={false}>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      selectable={true}
      selected={false}
      striped={false}>
      <TableHeaderColumn>
        Title
      </TableHeaderColumn>
      <TableHeaderColumn>
        Seniority
      </TableHeaderColumn>
      <TableHeaderColumn>
        Relations
      </TableHeaderColumn>
    </TableRow>
    <TableRow
      displayBorder={true}
      hoverable={false}
      hovered={false}
      selectable={true}
      selected={false}
      striped={false}>
      <TableRowColumn
        colSpan="3"
        hoverable={false}
        style={
          Object {
            "textAlign": "left",
          }
        }>
        Displaying 0 out of 0 Titles
      </TableRowColumn>
    </TableRow>
  </TableHeader>
  <TableBody
    allRowsSelected={false}
    deselectOnClickaway={false}
    displayRowCheckbox={true}
    multiSelectable={false}
    preScanRows={true}
    selectable={true}
    style={Object {}} />
</Table>
`;
